From 671419bde09d83aa51fa5068aaaa9fbeac183d6b Mon Sep 17 00:00:00 2001
From: Peter Marshall <pm@petermarshall.ca>
Date: Thu, 30 Jul 2026 13:16:34 -0400
Subject: [PATCH 14/16] media: ipu6: work around stream issues

The system's ring bus / RAM is clocked down during low power states,
causing flickering and FIFO timeut errors with some sensors when the CPU
is idle. This was originally noticed by Hans de Goede on the linux-media
mailing list (see https://marc.info/?l=linux-media&m=175999034702772).
Tighten the PM QoS value during IPU6 streaming to work around the issue.
A value of 50ms appears to work fine.

The IPU6 fails to start streaming with modes using 384 bytes per line or
less. Bump the minimum frame width up to a safe pixel limit.
---
 drivers/media/pci/intel/ipu6/ipu6-isys.c | 2 +-
 drivers/media/pci/intel/ipu6/ipu6-isys.h | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/pci/intel/ipu6/ipu6-isys.c b/drivers/media/pci/intel/ipu6/ipu6-isys.c
index 24db2763de54..ea8e87e5e915 100644
--- a/drivers/media/pci/intel/ipu6/ipu6-isys.c
+++ b/drivers/media/pci/intel/ipu6/ipu6-isys.c
@@ -98,7 +98,7 @@ enum ltr_did_type {
 	LTR_TYPE_MAX
 };
 
-#define ISYS_PM_QOS_VALUE	300
+#define ISYS_PM_QOS_VALUE	50
 
 static int isys_isr_one(struct ipu6_bus_device *adev);
 
diff --git a/drivers/media/pci/intel/ipu6/ipu6-isys.h b/drivers/media/pci/intel/ipu6/ipu6-isys.h
index 7fb8cb820912..ba616d1a905d 100644
--- a/drivers/media/pci/intel/ipu6/ipu6-isys.h
+++ b/drivers/media/pci/intel/ipu6/ipu6-isys.h
@@ -39,7 +39,8 @@ struct ipu6_bus_device;
 #define IPU6_ISYS_SIZE_PROXY_SEND_QUEUE 5
 #define IPU6_ISYS_NUM_RECV_QUEUE 1
 
-#define IPU6_ISYS_MIN_WIDTH		2U
+/* Work around stream failing with low bytes per line */
+#define IPU6_ISYS_MIN_WIDTH		392U
 #define IPU6_ISYS_MIN_HEIGHT		1U
 #define IPU6_ISYS_MAX_WIDTH		4672U
 #define IPU6_ISYS_MAX_HEIGHT		3416U
-- 
2.56.0

